### ISSUE TITLE
New build recommends wrong CPU config.

### ISSUE THREAD

**Issue by Josexv1** [fireice-uk/xmr-stak-cpu#87]
As you can see here the new recomendation CPU settings in the latest build, it gives bad config for non AES-NI CPUs

[2017-04-26 23:33:27] : Autoconf L3 size detected at 15360 KB.
[2017-04-26 23:33:27] : Autoconf core count detected as 2 on Linux.

**************** Copy&Paste ****************
"cpu_threads_conf" :
[
   { "low_power_mode" : true, "no_prefetch" : true, "affine_to_cpu" : 0 },
   { "low_power_mode" : true, "no_prefetch" : true, "affine_to_cpu" : 1 },
],
**************** Copy&Paste ****************

[2017-04-26 23:34:46] : Invalid thread confg - low_power_mode are unsupported on CPUs without AES-NI.

**Comment by psychocrypt** [comment]
Thanks for reporting, you found the first bug in the new release. We will fix it with the next release. 

Please change `low_power_mode` to false and maybe try to use more threads instead.

**Comment by fireice-uk** [comment]
Ah crap, forgot to remove an old guard _again_

**Comment by fireice-uk** [comment]
Fixed in 26c0134

**Comment by fireice-uk** [comment]
@Josexv1 If you are on Linux can you post the result of `cat /proc/cpuinfo` or if you are on Windows, make and model of your CPU so that I can verify that you really only have 2 cores?

**Comment by Josexv1** [comment]
I will post the different CPU I've tested, I have many VPS running it, for some it worked as expected, but for non AES-NI it recommend some true or false flags that doesn't work as they're not compatible with it.

On Thu, Apr 27, 2017, 9:25 AM fireice-uk <notifications@github.com<mailto:notifications@github.com>> wrote:

@Josexv1<https://github.com/Josexv1> If you are on Linux can you post the result of cat /proc/cpuinfo or if you are on Windows, make and model of your CPU so that I can verify that you really only have 2 cores?

**Comment by Josexv1** [comment]
Hello, this is the info from 2 of my VPSs i will test the fix.

CPU 1

"cpu_threads_conf" :
[
   { "low_power_mode" : true, "no_prefetch" : true, "affine_to_cpu" : 0 },
   { "low_power_mode" : true, "no_prefetch" : true, "affine_to_cpu" : 1 },
],
-----------------------------------------------------
CPUINFO

processor       : 0
vendor_id       : GenuineIntel
cpu family      : 6
model           : 63
model name      : Intel(R) Xeon(R) CPU E5-2620 v3 @ 2.40GHz
stepping        : 2
microcode       : 50
cpu MHz         : 1211.622
cache size      : 15360 KB
physical id     : 0
siblings        : 12
core id         : 0
cpu cores       : 6
apicid          : 0
initial apicid  : 0
fpu             : yes
fpu_exception   : yes
cpuid level     : 15
wp              : yes
flags           : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good xtopology nonstop_tsc aperfmperf cpuid_faulting pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer xsave avx f16c rdrand lahf_lm abm ida arat epb pln pts dtherm tpr_shadow vnmi flexpriority ept vpid fsgsbase bmi1 avx2 smep bmi2 erms invpcid cqm xsaveopt cqm_llc cqm_occup_llc
bogomips        : 4799.62
clflush size    : 64
cache_alignment : 64
address sizes   : 46 bits physical, 48 bits virtual
power management:

processor       : 1
vendor_id       : GenuineIntel
cpu family      : 6
model           : 63
model name      : Intel(R) Xeon(R) CPU E5-2620 v3 @ 2.40GHz
stepping        : 2
microcode       : 50
cpu MHz         : 1211.622
cache size      : 15360 KB
physical id     : 0
siblings        : 12
core id         : 1
cpu cores       : 6
apicid          : 2
initial apicid  : 2
fpu             : yes
fpu_exception   : yes
cpuid level     : 15
wp              : yes
flags           : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good xtopology nonstop_tsc aperfmperf cpuid_faulting pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer xsave avx f16c rdrand lahf_lm abm ida arat epb pln pts dtherm tpr_shadow vnmi flexpriority ept vpid fsgsbase bmi1 avx2 smep bmi2 erms invpcid cqm xsaveopt cqm_llc cqm_occup_llc
bogomips        : 4799.62
clflush size    : 64
cache_alignment : 64
address sizes   : 46 bits physical, 48 bits virtual
power management:
-----------------------------------------------------
HASHRATE REPORT
| ID | 2.5s |  60s |  15m | ID | 2.5s |  60s |  15m |
|  0 | (na) | 6.1 | 6.3 |  1 | (na) | 5.9 | 6.1 |
-----------------------------------------------------
Totals:   (na) 12.0 12.4 H/s
Highest:  16.6 H/s
-----------------------------------------------------

CPU 2

[2017-04-26 23:13:48] : Autoconf L3 size detected at 12288 KB.
[2017-04-26 23:13:48] : Autoconf core count detected as 5 on Linux.

**************** Copy&Paste ****************

"cpu_threads_conf" :
[
   { "low_power_mode" : true, "no_prefetch" : true, "affine_to_cpu" : 0 },
   { "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : 1 },
   { "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : 2 },
   { "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : 3 },
   { "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : 4 },
],
-----------------------------------------------------
processor       : 0
vendor_id       : GenuineIntel
cpu family      : 6
model           : 44
model name      : Intel(R) Xeon(R) CPU           E5620  @ 2.40GHz
stepping        : 2
microcode       : 21
cpu MHz         : 1919.902
cache size      : 12288 KB
physical id     : 0
siblings        : 8
core id         : 0
cpu cores       : 4
apicid          : 0
initial apicid  : 0
fpu             : yes
fpu_exception   : yes
cpuid level     : 11
wp              : yes
flags           : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good xtopology nonstop_tsc aperfmperf pni dtes64 monitor ds_cpl vmx smx est tm2 ssse3 cx16 xtpr pdcm pcid dca sse4_1 sse4_2 popcnt lahf_lm ida arat epb dtherm tpr_shadow vnmi flexpriority ept vpid
bogomips        : 4799.75
clflush size    : 64
cache_alignment : 64
address sizes   : 40 bits physical, 48 bits virtual
power management:

processor       : 1
vendor_id       : GenuineIntel
cpu family      : 6
model           : 44
model name      : Intel(R) Xeon(R) CPU           E5620  @ 2.40GHz
stepping        : 2
microcode       : 21
cpu MHz         : 1919.902
cache size      : 12288 KB
physical id     : 0
siblings        : 8
core id         : 1
cpu cores       : 4
apicid          : 2
initial apicid  : 2
fpu             : yes
fpu_exception   : yes
cpuid level     : 11
wp              : yes
flags           : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good xtopology nonstop_tsc aperfmperf pni dtes64 monitor ds_cpl vmx smx est tm2 ssse3 cx16 xtpr pdcm pcid dca sse4_1 sse4_2 popcnt lahf_lm ida arat epb dtherm tpr_shadow vnmi flexpriority ept vpid
bogomips        : 4799.75
clflush size    : 64
cache_alignment : 64
address sizes   : 40 bits physical, 48 bits virtual
power management:

processor       : 2
vendor_id       : GenuineIntel
cpu family      : 6
model           : 44
model name      : Intel(R) Xeon(R) CPU           E5620  @ 2.40GHz
stepping        : 2
microcode       : 21
cpu MHz         : 1919.902
cache size      : 12288 KB
physical id     : 0
siblings        : 8
core id         : 9
cpu cores       : 4
apicid          : 18
initial apicid  : 18
fpu             : yes
fpu_exception   : yes
cpuid level     : 11
wp              : yes
flags           : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good xtopology nonstop_tsc aperfmperf pni dtes64 monitor ds_cpl vmx smx est tm2 ssse3 cx16 xtpr pdcm pcid dca sse4_1 sse4_2 popcnt lahf_lm ida arat epb dtherm tpr_shadow vnmi flexpriority ept vpid
bogomips        : 4799.75
clflush size    : 64
cache_alignment : 64
address sizes   : 40 bits physical, 48 bits virtual
power management:

processor       : 3
vendor_id       : GenuineIntel
cpu family      : 6
model           : 44
model name      : Intel(R) Xeon(R) CPU           E5620  @ 2.40GHz
stepping        : 2
microcode       : 21
cpu MHz         : 1919.902
cache size      : 12288 KB
physical id     : 0
siblings        : 8
core id         : 10
cpu cores       : 4
apicid          : 20
initial apicid  : 20
fpu             : yes
fpu_exception   : yes
cpuid level     : 11
wp              : yes
flags           : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good xtopology nonstop_tsc aperfmperf pni dtes64 monitor ds_cpl vmx smx est tm2 ssse3 cx16 xtpr pdcm pcid dca sse4_1 sse4_2 popcnt lahf_lm ida arat epb dtherm tpr_shadow vnmi flexpriority ept vpid
bogomips        : 4799.75
clflush size    : 64
cache_alignment : 64
address sizes   : 40 bits physical, 48 bits virtual
power management:

processor       : 4
vendor_id       : GenuineIntel
cpu family      : 6
model           : 44
model name      : Intel(R) Xeon(R) CPU           E5620  @ 2.40GHz
stepping        : 2
microcode       : 21
cpu MHz         : 1919.902
cache size      : 12288 KB
physical id     : 1
siblings        : 8
core id         : 0
cpu cores       : 4
apicid          : 32
initial apicid  : 32
fpu             : yes
fpu_exception   : yes
cpuid level     : 11
wp              : yes
flags           : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good xtopology nonstop_tsc aperfmperf pni dtes64 monitor ds_cpl vmx smx est tm2 ssse3 cx16 xtpr pdcm pcid dca sse4_1 sse4_2 popcnt lahf_lm ida arat epb dtherm tpr_shadow vnmi flexpriority ept vpid
bogomips        : 4799.88
clflush size    : 64
cache_alignment : 64
address sizes   : 40 bits physical, 48 bits virtual
power management:
-----------------------------------------------------
HASHRATE REPORT
| ID | 2.5s |  60s |  15m | ID | 2.5s |  60s |  15m |
|  0 | (na) | 4.8 | (na) |  1 | 12.8 | 10.1 | (na) |
|  2 | (na) | 4.8 | (na) |  3 | 12.8 | 11.0 | (na) |
|  4 | (na) | 9.8 | (na) |
-----------------------------------------------------
Totals:   (na) 40.5 (na) H/s
Highest:  0.0 H/s

**Comment by Josexv1** [comment]
I have tested a new one, which i had with basic config with 230H/s
Now it has 300H/s

[2017-04-28 01:20:15] : Autoconf L3 size detected at 8192 KB.
[2017-04-28 01:20:15] : Autoconf core count detected as 8 on Linux.
```

"cpu_threads_conf" :
[
   { "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : 0 },
   { "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : 1 },
   { "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : 2 },
   { "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : 3 },
],
```


```
processor       : 0
vendor_id       : GenuineIntel
cpu family      : 6
model           : 58
model name      : Intel(R) Core(TM) i7-3770 CPU @ 3.40GHz
stepping        : 9
microcode       : 23
cpu MHz         : 3400.266
cache size      : 8192 KB
physical id     : 0
siblings        : 8
core id         : 0
cpu cores       : 4
apicid          : 0
initial apicid  : 0
fpu             : yes
fpu_exception   : yes
cpuid level     : 13
wp              : yes
flags           : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx rdtscp lm constant_tsc arch_perfmon pebs bts rep_good xtopology nonstop_tsc aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm ida arat epb xsaveopt pln pts dtherm tpr_shadow vnmi flexpriority ept vpid fsgsbase smep erms
bogomips        : 6800.53
clflush size    : 64
cache_alignment : 64
address sizes   : 36 bits physical, 48 bits virtual
power management:

```

**Comment by fireice-uk** [comment]
@Josexv1 did you ever see higher hashrates on the non-aes machines? The cpus look like low-power models.

**Comment by Josexv1** [comment]
In non-aes i think the max i ever saw was 40~45h/s but most of them are just between 20~30h/s

**Comment by YetAnotherRussian** [comment]
Oh, can u tell how to run the miner in config detection mode? Seems to be no description in config file or on github. Gonna join testing. Thanks.

**Comment by psychocrypt** [comment]
@YetAnotherRussian 

You need to substitute

```
"cpu_threads_conf" :
[ 
     { "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : 0 },
     { "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : 1 },
],
```
with
```
"cpu_threads_conf" : 
null,
```
to activate the auto suggestion.

@fireice-uk @YetAnotherRussian is right you added the old instead of the new config to the windows zip files, therefore autosuggestion is not used by default.

**Comment by Josexv1** [comment]
There's a chance that it may set the suggested CPU threads and configs automatically in the config.txt as soon as you run it? Like: ./bin/xmr-stak-cpu -> update the config.txt -> post that it have updated the config with the recommended values automatically ? that way will be easier for people like me having a bunch of cpus not SSHing and editting the file manually.

**Comment by Josexv1** [comment]
Also i have found another error with slow CPUs

[2017-04-27 03:59:13] : Autoconf failed: Couln't find L3 cache page.
[2017-04-27 03:59:13] : Autoconf failed: L3 size sanity check failed - 0 KB.
[2017-04-27 03:59:13] : Autoconf failed: Printing config for a single thread. Please try to add new ones until the hashrate slows down.

I can see the error happens with CPUs with only 1 core. Virtual CPU or just an old CPU.

lscpu
```
Architecture:          x86_64
CPU op-mode(s):        32-bit, 64-bit
Byte Order:            Little Endian
CPU(s):                1
On-line CPU(s) list:   0
Thread(s) per core:    1
Core(s) per socket:    1
Socket(s):             1
NUMA node(s):          1
Vendor ID:             GenuineIntel
CPU family:            6
Model:                 13
Model name:            QEMU Virtual CPU version (cpu64-rhel6)
Stepping:              3
CPU MHz:               2399.996
BogoMIPS:              4799.99
Hypervisor vendor:     KVM
Virtualization type:   full
L1d cache:             32K
L1i cache:             32K
L2 cache:              4096K
NUMA node0 CPU(s):     0
```

**Comment by fireice-uk** [comment]
Unfortunately that cpu doesn't have any official L3 cache so the result is "correct" - we will eventually expand it to more complicated setups like virtual cpus.

**Comment by fireice-uk** [comment]
The problems identified are fixed in the new release

**Comment by psychocrypt** [comment]
solved with #94

**Comment by katski11** [comment]
Hi Josexv1!

How did you achieve a Hash Rate of 300 H/s of an Intel Core i7-3770? What is the specs of your system? May I know how did you configure your config.txt in the xmr-stak-cpu? In my current setup, it only achieves a Hash Rate of 230 H/s on my Intel Core i7-3770 running on 3 Threads using Windows 7 x64 with 8GB memory on a single channel and a 120GB Transcend SSD. It kinda baffles me that it achieve such a high Hash Rate on that CPU.
